Framework/DataHandling/src/SaveOpenGenieAscii.cpp

```diff
@@ -49,337 +46,382 @@ void SaveOpenGenieAscii::init() {
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
+  const auto &y = m_inputWS->y(0);
+  for (const auto yVal : y) {
+    if (yCount % 10 == 0) {
+      yValsOutput += newlineStr;
     }
+    yCount++;
+    yValsOutput += std::to_string(yVal) + ' ';
+  }
 
-    else if (type.std::string::find("vector") &&
-             type.std::string::find("bool") != std::string::npos) {
-
-      auto tsp = ws->run().getTimeSeriesProperty<bool>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
+  const auto &e = m_inputWS->e(0);
+  for (const auto eVal : e) {
+    if (eCount % 10 == 0) {
+      eValsOutput += newlineStr;
     }
+    eCount++;
+    eValsOutput += std::to_string(eVal) + ' ';
+  }
 
-    else if (type.std::string::find("vector") &&
-             type.std::string::find("char") != std::string::npos) {
+  std::vector<std::tuple<std::string, int>> outDataStrings;
+  outDataStrings.push_back(std::make_tuple(xValsOutput, xCount));
+  outDataStrings.push_back(std::make_tuple(yValsOutput, yCount));
+  outDataStrings.push_back(std::make_tuple(eValsOutput, eCount));
+  return outDataStrings;
+}
 
-      auto tsp = ws->run().getTimeSeriesProperty<std::string>(name);
-      value = (tsp->lastValue());
-    }
+int SaveOpenGenieAscii::determineEnginXBankId() {
+  const auto &detectorIds = m_inputWS->getSpectrum(0).getDetectorIDs();
+  const std::string firstDetectorId = std::to_string(*detectorIds.cbegin());
 
-    if ((type.std::string::find("number") != std::string::npos) ||
-        (type.std::string::find("double") != std::string::npos) ||
-        (type.std::string::find("dbl list") != std::string::npos)) {
-      type = "Float";
-    }
+  if (firstDetectorId.length() != 6) {
+    throw std::runtime_error("Could not determine bank number as detector ID"
+                             " was not of expected length");
+  }
 
-    else if ((type.std::string::find("TimeValueUnit<bool>") !=
-              std::string::npos) ||
-             (type.std::string::find("TimeValueUnit<int>") !=
-              std::string::npos)) {
-      type = "Integer";
-    }
+  // ENGIN-X format is 1X001, 1X002, 1X003...etc. for detectors
+  // where X = 0 is bank 1. X = 1 is bank 2.
+  return firstDetectorId[1] == '0' ? 1 : 2;
+}
 
-    else if (type.std::string::find("string") != std::string::npos) {
-      type = "String";
+/** Reads the sample logs and converts them from the log name in
+  * Mantid to the expected log name in OpenGenie. If any names are
+  * unrecognised they are skipped. If the logs have multiple values
+  * the time weighted average is taken instead. These strings are
+  * then stored in the output buffer.
+  */
+void SaveOpenGenieAscii::getSampleLogs() {
+  // Maps Mantid log names -> Genie save file name / type
+  const std::unordered_map<std::string, std::tuple<std::string, std::string>>
+      MantidGenieNameMap = {
+          {"x", std::make_tuple("x_pos", m_floatType)},
```

Comment:
Using upper case for variable name. LLVM code guide actually states that this is what we should be doing, but this is inconsistent with the rest of Mantid.